agents/src/transcription.ts

```diff
@@ -118,12 +126,72 @@ export class BasicTranscriptionForwarder implements TranscriptionForwarder {
   }
 
   async close(interrupt: boolean): Promise<void> {
-    this.#isRunning = false;
+    this.isRunning = false;
 
     // Publish whatever we had as final
     if (!interrupt) {
-      this.currentCharacterIndex = this.#currentText.length;
+      this.currentCharacterIndex = this.currentText.length;
+    }
+    await this.publishTranscription(true);
+  }
+}
+
+export class ChatAndTranscriptionForwarder extends BasicTranscriptionForwarder {
```

Comment:
implementing this as a subclass feels unnecessarily brittle and hard to follow - it's already _slightly_ tricky code.